src/unitsConversion.ts

```diff
@@ -0,0 +1,290 @@
+ * @param optionsInput.pad - Whether to pad the fractional part with zeros.
+ * @param optionsInput.commify - Whether to add commas to separate thousands.
+ * @returns The string representation of the number.
+ * @throws Error if the number is invalid.
+ */
+export function fromWei(
+  weiInput: string | number | bigint,
+  unit: EthereumUnit,
+  optionsInput?: { pad?: boolean; commify?: boolean },
+) {
+  var wei = numberToBigInt(weiInput); // eslint-disable-line
+  var negative = wei < zero; // eslint-disable-line
+  const unitLower = unit.toLowerCase() as EthereumUnit;
+  const base = unitMapBigInt[unitLower];
+  const baseLength = unitLengths[unitLower];
+  const options = optionsInput ?? {};
+
+  if (base === undefined) {
+    throw new Error(
+      `[ethjs-unit] the unit provided ${unit} doesn't exists, please use the one of the following units ${JSON.stringify(
+        unitMap,
+        null,
+        2,
+      )}`,
+    );
+  }
+
+  // Handle special case of noether (base = 0)
+  if (base === zero) {
+    return negative ? '-0' : '0';
+  }
+
+  if (negative) {
+    wei = wei * negative1;
+  }
+
+  var fraction = (wei % base).toString(); // eslint-disable-line
+
+  fraction = fraction.padStart(baseLength, '0');
+
+  if (!options.pad) {
+    const fractionMatch = fraction.match(FRACTION_REGEX);
+    // istanbul ignore next: defensive fallback that's never reachable but necessary to satisfy TS
+    fraction = fractionMatch?.[1] ?? '0';
+  }
+
+  var whole = (wei / base).toString(); // eslint-disable-line
+
+  if (options.commify) {
+    whole = whole.replace(COMMIFY_REGEX, ',');
+  }
+
+  var value = `${whole}${fraction == '0' ? '' : `.${fraction}`}`; // eslint-disable-line
+
+  if (negative) {
+    value = `-${value}`;
+  }
+
+  return value;
+}
+
+/**
+ * Converts a number to Wei.
+ *
+ * @param etherInput - The number to convert to Wei.
+ * @param unit - The unit to convert to, default ether.
+ * @returns The number in Wei.
+ * @throws Error if the number is invalid.
+ */
+export function toWei(
+  etherInput: string | number | bigint,
+  unit: EthereumUnit,
+): bigint {
+  const unitLower = unit.toLowerCase() as EthereumUnit;
+  const base = unitMapBigInt[unitLower];
+  const baseLength = unitLengths[unitLower];
+
+  if (base === undefined) {
+    throw new Error(
+      `[ethjs-unit] the unit provided ${unit} doesn't exists, please use the one of the following units ${JSON.stringify(
+        unitMap,
+        null,
+        2,
+      )}`,
+    );
+  }
+
+  // Handle special case of noether (base = 0)
+  if (base === zero) {
+    return zero;
+  }
+
+  // Fast path for bigint inputs when unit is wei (no conversion needed)
+  if (typeof etherInput === 'bigint' && unitLower === 'wei') {
+    return etherInput;
+  }
+
+  // Fast path for bigint inputs with whole units (no fractional part)
+  if (typeof etherInput === 'bigint') {
+    return etherInput * base;
+  }
+
+  var ether = numberToString(etherInput); // eslint-disable-line
```

Comment:
I notice that in many places `var` is being used instead of `const` or `let`. I see that this was copied from the original, but maybe it's worth replacing those instances in this PR and remove the associated `eslint-disable` comments?